applications/StructuralMechanicsApplication/custom_elements/cr_beam_element_3D2N.cpp

```diff
@@ -1515,42 +1515,83 @@ void CrBeamElement3D2N::AddExplicitContribution(
     Variable<array_1d<double, 3>> &rDestinationVariable,
     const ProcessInfo &rCurrentProcessInfo) {
   KRATOS_TRY;
+
+  bounded_vector<double, msElementSize> damping_residual_contribution =
+      ZeroVector(msElementSize);
+  // calculate damping contribution to residual -->
+  if ((this->GetProperties().Has(RAYLEIGH_ALPHA) ||
+       this->GetProperties().Has(RAYLEIGH_BETA)) &&
+      (rDestinationVariable != NODAL_INERTIA)) {
+    Vector current_nodal_velocities = ZeroVector(msElementSize);
+    this->GetFirstDerivativesVector(current_nodal_velocities);
+    Matrix damping_matrix = ZeroMatrix(msElementSize, msElementSize);
+    ProcessInfo temp_process_information; // cant pass const ProcessInfo
+    this->CalculateDampingMatrix(damping_matrix, temp_process_information);
+    // current residual contribution due to damping
+    damping_residual_contribution =

```

Comment:
noalias
also do not set it to zero before, here you are assigning anyway

Reply:
noalias is ok but I need the zero initialisation because I am using it int the following if statements even if this line is not called